### PR TITLE
chore: DefaultRunFunc -> reconciler.Reconcile

### PR DESCRIPTION
**Pull request by karlkfi**
- Rename DefaultRunFunc as reconciler.Reconciler.
- Remove methods from the Reconciler interface only used by reconciler.Reconcile.
- Add Reconciler.UpdateSyncStatus to simplify event handling.
- Make reconciler methods private that no longer need to be exposed.

### CODE REVIEW

File: pkg/parse/reconciler.go [hunk 1]
```diff
@@ -83,23 +63,14 @@ func (p *reconciler) Options() *ReconcilerOptions {
...
 // ReconcilerState returns the current state of the reconciler.
 func (p *reconciler) ReconcilerState() *ReconcilerState {
 	return p.reconcilerState
 }
 
-// NewRootRunner creates a new runnable parser for parsing a Root repository.
-func NewRootRunner(recOpts *ReconcilerOptions, parseOpts *RootOptions) Reconciler {
+// NewClusterReconciler creates a new reconciler for reconciling cluster-scoped
+// and namespace-scoped resources configured with a RootSync.
+func NewClusterReconciler(recOpts *ReconcilerOptions, parseOpts *RootOptions) Reconciler {
```
Comment (sdowell) [inline pkg/parse/reconciler.go:73]: nit: I think `NewRootSyncReconciler` and `NewRepoSyncReconciler` may be more intuitive. `Cluster` and `Namespace` seem not as precise as they could be

Comment (karlkfi) [reply pkg/parse/reconciler.go:73]: I considered that. I'm not against it.

It seems to me that it's primarily reconciling the cluster objects (cluster-scoped and/or namespace-scoped) with the source objects, but as a controller it's also reconciling the RSync status with the spec...

Comment (karlkfi) [reply pkg/parse/reconciler.go:73]: But if that wasn't immediately obvious from the name, then I'm ok with your suggestion, which should be pretty clear.

Comment (karlkfi) [reply pkg/parse/reconciler.go:73]: I wish we could rename the resources themselves to be more clear, but that ship sailed before my time.

Comment (karlkfi) [reply pkg/parse/reconciler.go]: Renamed.

**Review by sdowell — APPROVED** [review]
/lgtm

**Pull request merged**